tests/oprint/expected/oprint.resi.txt

```diff
@@ -179,6 +179,8 @@ module Js: {
 }
 type arity0 = (.) => unit
 type arity0b = (.) => int
+type arity0c = Js.Fn.arity0<Js.Fn.arity0<array<int>>>
+type arity0d = (. unit) => unit
```

Comment:
This should be something like `(. unit) => unit => unit`
which btw is not currently possible anyway as it would currently format to ` (. unit, unit) => unit` which is arity 2.

Reply:
Not sure currently there's even a syntax that can express the intended type: uncurried function that returns a curried function

Reply:
Possibly related https://github.com/rescript-lang/rescript-compiler/issues/5395
so there might be something wrong already in the compiler when dealing with this case.
That, or there's something wrong just in error reporting (still in the compiler).